From e890a8068f64cf310bb22280391a6cb3e874da50 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 18 Nov 2022 22:14:04 +0000
Subject: [PATCH 2/2] gnu: Add cl-graph

* gnu/packages/lisp-xyz.scm (cl-graph, sbcl-cl-graph): New variables.
---
 gnu/packages/lisp-xyz.scm | 45 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 45 insertions(+)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index 58c97b5d68..fa0a1dfcaa 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -1976,6 +1976,51 @@ (define-public cl-hu.dwim.graphviz
 (define-public ecl-hu.dwim.graphviz
   (sbcl-package->ecl-package sbcl-hu.dwim.graphviz))
 
+(define-public sbcl-cl-graph
+  (let ((commit "3cb786797b24883d784b7350e7372e8b1e743508")
+        (revision "1"))
+    (package
+      (name "sbcl-cl-graph")
+      (version (git-version "0.10.2" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/gwkkwg/cl-graph")
+               (commit commit)))
+         (file-name (git-file-name "cl-graph" version))
+         (sha256
+          (base32 "1748rj52f2jmd5jvsy9jwhn0zf73sjzjhwhnljvq6yi2kdqr30kl"))))
+      (build-system asdf-build-system/sbcl)
+      (arguments
+       ;; TODO: (Sharlatan-20221118T215839+0000): Tests failed
+       ;;
+       ;; https://github.com/gwkkwg/cl-graph/issues/17
+       ;;
+       `(#:tests? #f
+         #:asd-systems '("cl-graph" "cl-graph+hu.dwim.graphviz")))
+      (native-inputs
+       (list sbcl-lift sbcl-moptilities))
+      (inputs
+       (list sbcl-hu.dwim.graphviz
+             sbcl-cl-mathstats
+             sbcl-dynamic-classes
+             sbcl-metabang-bind
+             sbcl-metacopy
+             sbcl-metatilities-base))
+      (home-page "https://github.com/gwkkwg/cl-graph")
+      (synopsis "Common Lisp library for manipulating graphs and running graph algorithms")
+      (description
+       "This package provides a Graph manipulation utilities for Common Lisp.")
+      (license license:expat))))
+
+;; NOTE: (Sharlatan-20221118T214734+0000): No ECL supoort
+;;
+;; ecl-cl-graph
+
+(define-public cl-cl-graph
+  (sbcl-package->cl-source-package sbcl-cl-graph))
+
 (define-public sbcl-babel
   ;; No release since 2014.
   (let ((commit "f892d0587c7f3a1e6c0899425921b48008c29ee3")
-- 
2.38.0

